newIDE/app/src/InstancesEditor/InstancesResizer.js

```diff
@@ -76,42 +82,90 @@ export default class InstancesResizer {
     this.totalDeltaX += deltaX;
     this.totalDeltaY += deltaY;
 
+    let hasRotatedInstance = false;
+    for (let i = 0; i < instances.length && !hasRotatedInstance; i++) {
+      const selectedInstance = instances[i];
+      hasRotatedInstance = selectedInstance.getAngle() !== 0;
+    }
+
     for (let i = 0; i < instances.length; i++) {
       const selectedInstance = instances[i];
 
       if (!selectedInstance.hasCustomSize()) {
         selectedInstance.setCustomWidth(
-          this.instanceMeasurer.getInstanceWidth(selectedInstance)
+          this.instanceMeasurer.getInstanceOBB(selectedInstance).width
         );
         selectedInstance.setCustomHeight(
-          this.instanceMeasurer.getInstanceHeight(selectedInstance)
+          this.instanceMeasurer.getInstanceOBB(selectedInstance).height
         );
       }
 
-      let initialSize = this.instanceSizes[selectedInstance.ptr];
-      if (!initialSize) {
-        initialSize = this.instanceSizes[selectedInstance.ptr] = {
-          width: selectedInstance.getCustomWidth(),
-          height: selectedInstance.getCustomHeight(),
+      let initialOBB = this.instanceOBBs[selectedInstance.ptr];
+      if (!initialOBB) {
+        initialOBB = new Rectangle();
+        initialOBB = this.instanceMeasurer.getInstanceOBB(
+          selectedInstance,
+          initialOBB
+        );
+        this.instanceOBBs[selectedInstance.ptr] = initialOBB;
+      }
+      let initialAABB = this.instanceAABBs[selectedInstance.ptr];
+      if (!initialAABB) {
+        initialAABB = new Rectangle();
+        initialAABB = this.instanceMeasurer.getInstanceAABB(
+          selectedInstance,
+          initialAABB
+        );
+        this.instanceAABBs[selectedInstance.ptr] = initialAABB;
+      }
+      let initialPosition = this.instancePositions[selectedInstance.ptr];
+      if (!initialPosition) {
+        initialPosition = this.instancePositions[selectedInstance.ptr] = {
+          x: selectedInstance.getX(),
+          y: selectedInstance.getY(),
         };
       }
 
       selectedInstance.setHasCustomSize(true);
-      selectedInstance.setCustomWidth(
-        this._roundWidth(
-          initialSize.width + this._getSizeDeltaX(proportional, initialSize)
-        )
-      );
-      selectedInstance.setCustomHeight(
-        this._roundHeight(
-          initialSize.height + this._getSizeDeltaY(proportional, initialSize)
-        )
-      );
+      if (hasRotatedInstance) {
```

Comment:
Do you think we could avoid this special case when there are rotated instances? Could you describe what changes when there are rotated instances? :)

Reply:
When there is a rotated instance, the scaling will keep the proportions. It's to avoid shear transformations as it's not supported by the model. I will add a comment.
![Shear](https://user-images.githubusercontent.com/2611977/122649702-4cf93280-d12f-11eb-8df2-e6c43ae0b0d0.png)